pkg/execution/driver/driver.go

```diff
@@ -29,39 +32,57 @@ type Driver interface {
 // MarshalV1 marshals state as an input to driver runtimes.
 func MarshalV1(
 	ctx context.Context,
-	s state.State,
+	sl sv2.StateLoader,
+	md sv2.Metadata,
 	step inngest.Step,
 	stackIndex int,
 	env string,
 	attempt int,
 ) ([]byte, error) {
-	md := s.Metadata()
 
 	req := &SDKRequest{
-		Events:  s.Events(),
-		Event:   s.Event(),
-		Actions: s.Actions(),
+		Event:   map[string]any{},
+		Events:  []map[string]any{},
```

Comment:
should this be just `[]event.Event` instead?
don't know if we need to serialize it to JSON bytes if the HTTP requester is gonna do it for us anyways.

Reply:
I'm leaving this as-is due to the gateway.  You're right that drivers could do with a major refactor.